### PR TITLE
Make install: use libraries instead of .la and switch to standard *.so naming

### PR DESCRIPTION
**Pull request by cburstedde**
**_Describe your changes here:_**

I've worked on the make-install build. This branch does not require any .la files in this mode.
Try configuring --with-sc= and --with-p4est=, where both have been make installed before.
I'm adding an rpath to the t8code executables for convenience so they run from anywhere.

The submodules are committed but not versioned at this point. Will do when approved.
The standard build without configure options should work unchanged.

Comments?

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [ ] The author added a BSD statement to `doc/` (or already has one)
- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] New Datatypes are added to t8indent_custom_datatypes.txt
- [ ] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

### CODE REVIEW

**Comment by jmark** [comment]
@cburstedde When I compile this branch, with and without `--with-{sc,p4est}` then `libt8.so` does not point to `libsc.so` and `libp4est.so`. Is this on purpose?

**Comment by cburstedde** [comment]
> @cburstedde When I compile this branch, with and without `--with-{sc,p4est}`, then `libt8.so` does not point to `libsc.so` and `libp4est.so`. Is this on purpose?

Yes, I decided to not give any -L and -l flags to building the library itself. What do you think is the better convention?
This mechanism does not work on half the existing platforms anyway according to the libtool documentation.

**Comment by jmark** [comment]
For the Julia wrapper we run into problems, since the wrapper builder does some linking checks directly on `libt8.so`. They fail since `libp4est.so` and `libsc.so` are not loaded.

**Comment by cburstedde** [comment]
So maybe it would make sense to add -L and -l of dependencies to the LIBADD of the higher-level build.
I can try this and see if that solves these problems.

**Comment by jmark** [comment]
When this issue https://github.com/cburstedde/p4est/issues/204 is resolved, we can look at this PR again.

@cburstedde

**Comment by jmark** [comment]
The current state of this PR fixes the issues raised in https://github.com/DLR-AMR/t8code/issues/483 and https://github.com/DLR-AMR/t8code/issues/496.

We should wait for the merge till https://github.com/cburstedde/libsc/pull/114 (libsc) and https://github.com/cburstedde/p4est/pull/208 (p4est) are merged into the main branches.

@cburstedde @sloede

File: src/Makefile.am [hunk 1]
```diff
@@ -132,10 +132,14 @@ src_libt8_la_SOURCES = \
 src_libt8_la_CPPFLAGS = $(AM_CPPFLAGS) $(T8_CPPFLAGS)
 ## This is the official API versioning scheme of libtool.  Please see:
 ## Read https://www.gnu.org/software/libtool/manual/libtool.html#Versioning
-src_libt8_la_LDFLAGS = -release $(VERSION) -version-info 2:0:0
+src_libt8_la_LDFLAGS = -version-info 2:0:0
 src_libt8_la_LIBADD = @T8_P4EST_LDADD@ @T8_SC_LDADD@
```
Comment (cburstedde) [inline src/Makefile.am:136]: This works by accident; with the latest commits on feature-make-install you can use T8_*_LIBADD.

Comment (jmark) [reply src/Makefile.am:136]: Can you elaborate respectively propose a patch? I'm not savvy in autotools and coded by trial-and-error only.

**Review by sloede — COMMENTED** [review]
I'm not an expert on autotools, but with the proposed change by Carsten this LGTM

**Comment by jmark** [comment]
@cburstedde 

The t8code tests on Github CI fails with:
```
g++: error: T8_P4EST_LIBADD@: No such file or directory
g++: error: T8_SC_LIBADD@: No such file or directory
```

On my local machine and on [Yggdrasil ](https://github.com/JuliaPackaging/Yggdrasil/pull/6565)all compiles just fine.

Have you seen this behavior before? Any clue what the problem is? Incompatible or old autotools version on GH?

**Comment by jmark** [comment]
Ok, found the problem. The submodules for `sc` and `p4est` were not pointing to https://github.com/DLR-AMR/t8code/issues/483 and https://github.com/DLR-AMR/t8code/issues/496.

@cburstedde

**Comment by jmark** [comment]
With this PR I could finally register an official [t8code_jll](https://juliahub.com/ui/Packages/t8code_jll/6PmXJ/1.1.2+2) Julia package. But this is only a temporary solution, since this builds on pre-releases and branches of sc, p4est and t8code. If this PR will be merged in the near future, the package could be build upon official releases of the three libraries.

**Comment by cburstedde** [comment]
Once the latest develop branches of p4est and sc are made to master, these will have to be updated here to work together with t8code.

**Comment by jmark** [comment]
This PR got obsolete since the proposed changes have been included in other PRs already. Many thanks @cburstedde for the input. This PR was nevertheless vital in the understanding of the issue. Gonna close this PR now.

**Pull request closed without merging**